shared/message/input/PS3Controller.proto

```diff
@@ -0,0 +1,246 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+syntax = "proto3";
+
+package message.input.PS3Controller;
+
+import "google/protobuf/timestamp.proto";
+
+message CrossButton {
+    // The timestamp of the event
+    google.protobuf.Timestamp timestamp = 1;
+
+    // The state of the button.
+    bool pressed = 2;
+
+    // Is this an initialisation message?
+    bool init = 3;
```

Comment:
Do we even need to know about the init messages? Are they useful outside of the module that creates them?

Reply:
Init messages are the initial state of the system. The controller sends these (one for each button) when it connects I think

Reply:
So you get an initial value for each button type, but since the person didn't press the button manually you want to know this was an "initialisation" message. It still might be worthwhile ignoring the init messages as this is basically a "controller connected" message

Reply:
it may be important to be able to differentiate between the initial state of the controller and a future state, i think it should remain